pkg/server/controller.go

```diff
@@ -145,6 +282,264 @@ type indexPage struct {
 	BaseURL       string
 }
 
+func (ctrl *Controller) decodeGoogleCallbackResponse(resp *http.Response) (name string, err error) {
```

Comment:
Just a NB: I myself like naked return, but nowadays almost nobody uses this (except for deferred recovery use case). Perhaps we should do so as well.